core/app/models/spree/variant/options_presentation.rb

```diff
@@ -0,0 +1,47 @@
+class Spree::Variant::OptionsPresentation
+  WORDS_CONNECTOR = ', '.freeze
+
+  attr_reader :variant
+
+  def initialize(variant)
+    @variant = variant
+  end
+
+  def to_sentence
+    options = option_values
+    options = sort_options(options)
+    options = present_options(options)
+
+    join_options(options)
+  end
+
+  private
+
+  def option_values
+    variant.option_values.includes(:option_type).to_a
```

Comment:
IMO, it'd be better to rely on the outer scope to preload the relations (the relations can be used somewhere else).